### DISCUSSION TITLE
unique_ptr

### DISCUSSION THREAD

**Discussion by PuppyRush** [smc-tech-talk/Lets-Trade#67] [Feedback]
to @Bundy-Mundi 
가능하다면 지난번에 해결못한 unique_ptr 을 좀 다시 보고 컴파일 에러 원인과 해결방법을 공유하고자 합니다. 저도 그 부분을 잘 모르고 있어서요.
fork했지만 빌드가 아직 준비 안되어 있는것으로 보였습니다. 빌드가 준비되거나 혹은 현재 이 논의를 진행하기 어려우면 말씀해주세요 =)

**Comment by ben9543** [comment]
@PuppyRush  

안녕하세요! 저번에 만나뵙고 C++ 문법위주로 개인공부만 진행중이라 따로 이 플젝에 많이 해놓은게 없어서 조금 민망하네요 😅 혹시 코드가 좀 이상하다 느껴지시면 코드 fork 하구 `dev` 브랜치로 바꿔주세요! `master` 브랜치는 업데이트가 안된지 조금 오래됐어요(준비되면 합치려다가 계속 안하게 되는...)

저번에 아마 같이 하다가 안되었던게  `portfolio.cpp` 파일의

```cpp
Portfolio::Portfolio(const vector<std::unique_ptr<Stock>>& stocks){
    InitializeEmptyShares(stocks);
};
void Portfolio::InitializeEmptyShares(const vector<std::unique_ptr<Stock>>& stocks){
    try{
        for(auto& s : stocks){
            Share share = Share(s.get(), 0);
            this->shares.push_back(share);
        }
    } catch(const char* errMessage){
        exit(1);
    }
};
```

이부분 이었던 것 같은데 (제가 틀렸으면 말씀해주세요 😊) 일단 지금은 그때 같이 `Portfolio::Portfolio(const vector<std::unique_ptr<Stock>>& stocks)`  이 부분에 `const`  추가 후 잘 작동중이에요.  제가 혹시 기억못하는 저희가 그냥 넘어갔던 부분이 있었다면 꼭 말씀해 주세요!

CMake를 통한 빌드는 제가 확실히 이해하고 쓰고 있지 않아서  지금 사용하지 못하고 있어요 :(  g++ 컴파일러에 .cpp 파일들을 전부 추가해주는 방식으로 테스트만 해보고있습니다.

`g++ -o start portfolio.cpp player.cpp share.cpp stock.cpp transaction.cpp csvExtractor.cpp main.cpp`

부족한 부분이 많고 정신없는 repo에 신경써주셔서 정말 감사합니다 😥
